src/main/java/org/springframework/data/couchbase/config/AbstractCouchbaseConfiguration.java

```diff
@@ -310,6 +317,44 @@ public ObjectMapper couchbaseObjectMapper() {
 		return mapper;
 	}
 
+	/**
+	 * The default blocking transaction manager. It is an implementation of CallbackPreferringTransactionManager
+	 * CallbackPreferrringTransactionmanagers do not play well with test-cases that rely
+	 * on @TestTransaction/@BeforeTransaction/@AfterTransaction
+	 *
+	 * @param clientFactory
+	 * @return
+	 */
+	@Bean(BeanNames.COUCHBASE_TRANSACTION_MANAGER)
+	CouchbaseCallbackTransactionManager callbackTransactionManager(CouchbaseClientFactory clientFactory) {
```

Comment:
should the name of this bean be couchbaseTransactionManager? (the callback in here, at least in the bean name, does not feel right..) (leaks impl detail and gives no hint to what actual transaction manager it is -- and the bean is also named differently)

Reply:
I'll change the method name.
The actual bean name is already couchbaseTransactionManager.
 COUCHBASE_TRANSACTION_MANAGER = "couchbaseTransactionManager";